os_migrate/tests/unit/test_resource.py

```diff
@@ -280,8 +280,9 @@ def test_dst_prerequisites_errors(self):
         res = FakeResource.from_data(valid_fakeresource_data())
         self.assertEqual(res.dst_prerequisites_errors(None), [])
 
-        def exception_refs_from_ser(_self, conn, filters=None):
+        def exception_refs_from_ser(_self, conn):
             raise ResourceFailure("Image 'asdf' not found.")
-        res._refs_from_ser = exception_refs_from_ser
+        # from a function to a bound method via __get__
+        res._refs_from_ser = exception_refs_from_ser.__get__(res, res.__class__)
```

Comment:
Took me a bit to find what was wrong here... The function needs to be bound to the instance via __get__, shouldn't be just assigned as-is. Previously it worked by accident because of the last defaulted parameter, but the parameters were erroneously shifted by 1 when actually calling the method. Removing the filters parameter exposed this.